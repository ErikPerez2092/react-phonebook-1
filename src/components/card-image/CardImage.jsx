import React from 'react';
import FriendsImage from './friends.jpeg';

const CardImage = () => <img alt="friends" src={FriendsImage} />;

export default CardImage;
