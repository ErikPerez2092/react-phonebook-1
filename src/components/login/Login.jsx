import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';

import GoogleSvg from '../google-svg';
import CardImage from '../card-image/CardImage';
import FriendsImage from './friends.jpeg';
import './login.css';

const Login = () => (
  <section className="Login">
    <CardImage title="react-phonebook!" image={FriendsImage} />
    <div className="Login-google">
      <GoogleSvg />
      <RaisedButton label="Login" />
    </div>
  </section>
);

export default Login;
