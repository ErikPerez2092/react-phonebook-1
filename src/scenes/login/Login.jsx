import React, { Component } from 'react';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';
import firebase from 'firebase';
import CircularProgress from 'material-ui/CircularProgress';
import { withRouter } from 'react-router-dom';

import { GoogleSvg, CardImage } from '../../components';
import FriendsImage from './friends.jpeg';
import './login.css';

class Login extends Component {
  constructor() {
    super();
    this.state = {
      isLoading: false,
      isDialogOpen: false,
      errorMessage: null,
    };
  }

  handleDialogOpen = () => {
    this.setState({ isDialogOpen: true });
  };

  handleDialogClose = () => {
    this.setState({ isDialogOpen: false });
  };

  handleAuth = () => {
    this.setState({ isLoading: true });
    const provider = new firebase.auth.GoogleAuthProvider();

    /* eslint react/prop-types: 0 */
    firebase
      .auth()
      .signInWithPopup(provider)
      .then(() => this.props.history.push('/contacts'))
      .catch(err => {
        this.setState({ isLoading: false });
        this.handleDialogOpen();
        this.setState({ errorMessage: err.message });
      });
  };

  render() {
    const action = <FlatButton label="OK" onClick={this.handleDialogClose} />;
    const isLoading = this.state.isLoading ? (
      <div className="u-progress">
        <CircularProgress size={80} thickness={5} />
      </div>
    ) : null;
    return (
      <section className="Login">
        {isLoading}
        <CardImage title="react-phonebook!" image={FriendsImage} />
        <div className="Login-google">
          <GoogleSvg />
          <RaisedButton label="Login" onClick={this.handleAuth} />
          <Dialog actions={action} modal={false} open={this.state.isDialogOpen} onRequestClose={this.handleDialogClose}>
            {this.state.errorMessage}
          </Dialog>
        </div>
      </section>
    );
  }
}

export default withRouter(Login);
