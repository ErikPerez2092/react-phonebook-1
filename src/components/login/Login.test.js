import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import { shallow } from 'enzyme';
import Login from './Login';
import CardImage from '../card-image';
import GoogleSvg from '../google-svg';

describe('LoginComponent', () => {
  it('should render <Login />', () => {
    const loginComponent = shallow(<Login />);
    expect(loginComponent.getElements()).toMatchSnapshot();
  });
  it('should have <CardImage />', () => {
    const loginComponent = shallow(<Login />);
    expect(loginComponent.find(CardImage)).toHaveLength(1);
  });
  it('should have <GoogleSvg />', () => {
    const loginComponent = shallow(<Login />);
    expect(loginComponent.find(GoogleSvg)).toHaveLength(1);
  });
  it('should have <RaisedButton />', () => {
    const loginComponent = shallow(<Login />);
    expect(loginComponent.find(RaisedButton)).toHaveLength(1);
  });
});
