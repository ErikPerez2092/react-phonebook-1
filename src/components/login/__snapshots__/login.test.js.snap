// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent should render <Login /> 1`] = `
Array [
  <section
    className="Login"
>
    <CardImage
        image="friends.jpeg"
        title="react-phonebook!"
    />
    <div
        className="Login-google"
    >
        <GoogleSvg />
        <RaisedButton
            disabled={false}
            fullWidth={false}
            label="Login"
            labelPosition="after"
            primary={false}
            secondary={false}
        />
    </div>
</section>,
]
`;
